### PR TITLE
Close chat

### PR DESCRIPTION
**Pull request by piehei**
This PR implements the closing of a chat between two users.

Notably:

* new endpoint /api/matches/close (accepts a POST)
* body of the POST documented in README
* new firebase function which handles the actual closing of the chat

How is the chat closed in practice? The function adds a new KV pair "closed": true into the chat object itself. Firebase security rules themselves control the read/write permissions of a particular chat so that if the value is closed, you cannot write anything to the chat.

(Note: the Firebase security rules will be provided out-of-git-band in a json format.)

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -143,3 +143,25 @@ exports.sendPushMessage = functions.database.ref('/chats/{chatId}/')
     })
   });
 
+// this writes "closed": true to chats/chatId and so closes the chat
+// for writing. reading will still be allowed.
+exports.closeChatId = functions.https.onRequest((req, res) => {
+  
+  if (req.get('FUNCTION_SECRET_KEY') !== functions.config().functions.secret) {
+    console.log('not authenticated, FUNCTION_SECRET_KEY header is barps');
```
Comment (kaupunki-apina) [inline functions/index.js:151]: What does "barps" stand for? :D

File: src/core/function-core.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
 const process = require('process');
 const fetch = require('node-fetch');
 
-const URL_CREATE_CHAT = "https://us-central1-whappu-183808.cloudfunctions.net/addNewChatBetweenUsers";
-const URL_ADD_PUSHTOKEN = "https://us-central1-whappu-183808.cloudfunctions.net/addPushTokenForUserId";
+const BASE_URL = "https://us-central1-whappu-183808.cloudfunctions.net";
```
Comment (kaupunki-apina) [inline src/core/function-core.js]: I still recommend turning this into environment variable

Comment (piehei) [reply src/core/function-core.js]: I will. Thanks.

File: functions/index.js (hunk 1 above)
Comment (piehei) [reply functions/index.js:151]: :D

**Pull request merged**